sensor_msgs/include/sensor_msgs/point_field_conversion.hpp

```diff
@@ -0,0 +1,207 @@
+{
+  typedef typename pointFieldTypeAsType<point_field_type>::type type;
+  return static_cast<T>(*(reinterpret_cast<type const *>(data_ptr)));
+}
+
+/*!
+ * \Converts a value at the given pointer position interpreted as the datatype
+ *  specified by the given datatype parameter to the given template type and returns it.
+ * \param data_ptr    pointer into the point cloud 2 buffer
+ * \param datatype    sensor_msgs::PointField datatype value
+ * \tparam T          return type
+ */
+template<typename T>
+inline T readPointCloud2BufferValue(const unsigned char * data_ptr, const unsigned char datatype)
+{
+  switch (datatype) {
+    case sensor_msgs::msg::PointField::INT8:
+      return readPointCloud2BufferValue<sensor_msgs::msg::PointField::INT8, T>(data_ptr);
+    case sensor_msgs::msg::PointField::UINT8:
+      return readPointCloud2BufferValue<sensor_msgs::msg::PointField::UINT8, T>(data_ptr);
+    case sensor_msgs::msg::PointField::INT16:
+      return readPointCloud2BufferValue<sensor_msgs::msg::PointField::INT16, T>(data_ptr);
+    case sensor_msgs::msg::PointField::UINT16:
+      return readPointCloud2BufferValue<sensor_msgs::msg::PointField::UINT16, T>(data_ptr);
+    case sensor_msgs::msg::PointField::INT32:
+      return readPointCloud2BufferValue<sensor_msgs::msg::PointField::INT32, T>(data_ptr);
+    case sensor_msgs::msg::PointField::UINT32:
+      return readPointCloud2BufferValue<sensor_msgs::msg::PointField::UINT32, T>(data_ptr);
+    case sensor_msgs::msg::PointField::FLOAT32:
+      return readPointCloud2BufferValue<sensor_msgs::msg::PointField::FLOAT32, T>(data_ptr);
+    case sensor_msgs::msg::PointField::FLOAT64:
+      return readPointCloud2BufferValue<sensor_msgs::msg::PointField::FLOAT64, T>(data_ptr);
+  }
+  // This should never be reached, but return statement added to avoid compiler warning. (#84)
+  return T();
+}
+
+/*!
+ * \Inserts a given value at the given point position interpreted as the datatype
+ *  specified by the template argument point_field_type.
+ * \param data_ptr            pointer into the point cloud 2 buffer
+ * \param value               the value to insert
+ * \tparam point_field_type   sensor_msgs::PointField datatype value
+ * \tparam T                  type of the value to insert
+ */
+template<int point_field_type, typename T>
+inline void writePointCloud2BufferValue(unsigned char * data_ptr, T value)
+{
+  typedef typename pointFieldTypeAsType<point_field_type>::type type;
+  *(reinterpret_cast<type *>(data_ptr)) = static_cast<type>(value);
+}
+
+/*!
+ * \Inserts a given value at the given point position interpreted as the datatype
+ *  specified by the given datatype parameter.
+ * \param data_ptr    pointer into the point cloud 2 buffer
+ * \param datatype    sensor_msgs::PointField datatype value
+ * \param value       the value to insert
+ * \tparam T          type of the value to insert
+ */
+template<typename T>
+inline void writePointCloud2BufferValue(
```

Comment:
This seems like an odd function; it doesn't seem to be used anywhere, but it "feels" like an internal function, meant to be used by the functions in `point_cloud_conversion.h`.  Should we just remove it and port it if someone asks?

Reply:
Yes, it wasn't clear to me where it was being used, I'm okay with removing it for now.  A quick github search on that method name seems to be all copies of this file or similar.

Reply:
Removed.